checks/dangerous_workflow.go

```diff
@@ -281,20 +277,34 @@ func workflowUsesCodeCheckoutAndNoEnvironment(workflow *actionlint.Workflow,
 		return false
 	}
 
-	_, usesPullRequest := triggers[triggerPullRequest]
-	_, usesPullRequestTarget := triggers[triggerPullRequestTarget]
-
 	for _, job := range workflow.Jobs {
-		chk, ref := jobUsesCodeCheckout(job)
-		if ((chk && usesPullRequest) ||
-			(chk && usesPullRequestTarget && strings.Contains(ref, checkoutUntrustedRef))) &&
-			!jobUsesEnvironment(job) {
-			return true
-		}
+        if jobUsesCodeCheckoutAndNoEnvironment(job, triggers) {
+            return true
+        }
 	}
 	return false
 }
 
+func jobUsesCodeCheckoutAndNoEnvironment(job *actionlint.Job, triggers map[triggerName]bool,
+) bool {
+	if job == nil {
+		return false
+	}
+	_, usesPullRequest := triggers[triggerPullRequest]
+	_, usesPullRequestTarget := triggers[triggerPullRequestTarget]
+	_, usesWorkflowRun := triggers[triggerWorkflowRun]
+
+    chk, ref := jobUsesCodeCheckout(job)
+    if ((chk && usesPullRequest) ||
```

Comment:
for readability, I'd go with:
for the sake of readability, I'd break it to:
```
if !jobUsesEnvironment(job) {
  if rest-of-the-checks
}
```

Reply:
wait, in the original code, the env check was sufficient to return nil in the first usage, but used && for the second usage.
isn't that a "breaking" change ?

Reply:
The test checks for unsafe checkout and no environment restriction, the original code check for the second and then for the first. 
both should be logical equivalent